tools/src/launcher/Main.hs

```diff
@@ -128,12 +129,20 @@ data NodeData = NodeData
     , ndArgs    :: ![Text]
     , ndLogPath :: Maybe FilePath }
 
--- | Updater path, args, windows runner path, archive path
-data UpdaterData = UpdaterData
-    { udPath        :: !FilePath
-    , udArgs        :: ![Text]
-    , udWindowsPath :: Maybe FilePath
-    , udArchivePath :: Maybe FilePath
+data UpdaterData = WindowsUpdaterData WinUD | OSXUpdaterData OSXUD | UnsupportedOSUpdateData UnsupOSUD
+
+data UnsupOSUD = UnsupOSUD
+
+data WinUD = WinUD 
+    { wudExePath :: !FilePath -- Installer.exe
+    , wudArgs    :: ![Text]
+    , wudBatPath :: !FilePath -- Installer.bat
```

Comment:
To be honest I don't like that more knowledge about update process is added to launcher. This knowledge should be mostly in installers ([daedalus/installers](https://github.com/input-output-hk/daedalus/tree/develop/installers)).
Some knowledge was here for a long time (about bat usage on windows), so I am not asking to eliminate. But let's describe these fields using a bit more abstract terms.
`wudExePath` – installer executable (e. g. Installer.exe).
`wudArgs` – command line arguments passed to the installer executable.
`wudBatPath` – path where bat file with installer instructions will be written.